src/components/version-redirect.jsx

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import {Redirect} from 'react-router-dom';
+
+export default class VersionRedirect extends React.Component {
+  constructor (props) {
+    super (props);
+    this.state = {
+      pageId: null
+    };
+  }
+
+  componentDidMount () {
+    const versionId = this.props.match.params.versionId;
+    this.props.api.getVersionById(versionId).then(data => {
+      this.setState({pageId: data.page_uuid});
+    });
```

Comment:
It might be simpler to just call `this.props.history.replace()` instead of setting state here, e.g:

```js
this.props.api.getVersionById(versionId).then(data =>
  this.props.history.replace(`/page/${data.page_uuid}/..${versionId}`));
```

Then you don’t need the constructor at all and the `render()` method only needs to always return `null` because there’s no state to track.

(Side note: it’s pretty edge-case-y, but you could check that the `versionId` in the props still matches the one you initially had before redirecting… in case the request was really slow and the user did something to navigate to a new version in the mean time.)

Reply:
Cool, will change. I didn't know I could do that with `history`. I thought I always needed a `<Redirect />`. If all you have is a hammer...

Reply:
Yeah. If you didn’t have to make an async API call, the `<Redirect>` would probably be clearer and more declarative.

Now that I’m thinking more about it, though… maybe you do need to have some state here so that you can render:

- The loading indicator while checking for the version data,
- An error/not found message if the version ID is bogus

…at that point, maybe the `<Redirect>` is just fine ¯\\\_(ツ)\_/¯